languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -0,0 +1,244 @@
+attribute pop_scoped_symbol = symbol    => type = "pop_scoped_symbol", symbol = symbol
+attribute pop_symbol = symbol           => type = "pop_symbol", symbol = symbol
+attribute push_node = node              => type = "push_symbol", node_symbol = node
+attribute push_scoped_node = node       => type = "push_scoped_symbol", node_symbol = node
+attribute push_scoped_symbol = symbol   => type = "push_scoped_symbol", symbol = symbol
+attribute push_symbol = symbol          => type = "push_symbol", symbol = symbol
+attribute scoped_node_definition = node => type = "pop_scoped_symbol", node_symbol = node, is_definition
+attribute scoped_node_reference = node  => type = "push_scoped_symbol", node_symbol = node, is_reference
+attribute symbol_definition = symbol    => type = "pop_symbol", symbol = symbol, is_definition
+attribute symbol_reference = symbol     => type = "push_symbol", symbol = symbol, is_reference
+
+attribute node_symbol = node            => symbol = (source-text node), source_node = node
+
+; ######                                                   
+; #     # #####   ####   ####  #####    ##   #    #  ####  
+; #     # #    # #    # #    # #    #  #  #  ##  ## #      
+; ######  #    # #    # #      #    # #    # # ## #  ####  
+; #       #####  #    # #  ### #####  ###### #    #      # 
+; #       #   #  #    # #    # #   #  #    # #    # #    # 
+; #       #    #  ####   ####  #    # #    # #    #  ####  
+;
+; ########################################################
+
+(program)@prog {
+  node @prog.defs
+  node @prog.lexical_scope
+  edge @prog.lexical_scope -> ROOT_NODE
+  edge ROOT_NODE -> @prog.defs
+}
+
+(program (_)@declaration)@prog {
+  edge @prog.defs -> @declaration.defs
+  edge @declaration.lexical_scope -> @prog.lexical_scope
+}
+
+
+;; Class Expressions
+
+(class_declaration
+  name: (identifier) @name
+  body: (class_body) @class_body) @class {
+
+  node @name.pop
+  node @class.lexical_scope
+  node @class.defs
+  attr (@name.pop) node_definition = @name
+
+  edge @class_body.lexical_scope -> @class.lexical_scope
+
+  edge @class_body.lexical_scope -> @class_body.defs
+  attr (@class_body.lexical_scope -> @class_body.defs) precedence = 1
+  edge @class.defs -> @name.pop
+  edge @name.pop -> @class_body.defs
+}
+
+(class_declaration
+  superclass: (superclass
+    (type_identifier) @identifier) @superclass
+    body: (class_body) @class_body) @class {
+
+    node @identifier.ref
+    attr (@identifier.ref) node_reference = @identifier
+
+    edge @identifier.ref -> @class.lexical_scope
+    edge @class_body.lexical_scope -> @identifier.ref
+}
+
+(class_body) @class_body {
+  node @class_body.lexical_scope
+  node @class_body.defs
+}
+
+(class_body (_)@declaration)@class_body {
+  edge @class_body.defs -> @declaration.defs
+  edge @declaration.lexical_scope -> @class_body.lexical_scope
+}
+
+(method_invocation
+  name: (identifier) @method_name) @method_invocation {
+
+  node @method_name.value
```

Comment:
This is fair, I started adding this in for a new test case I was working on that uses `@method_name.value` in a different stanza. I'll remove it for now though in case I can work around it